### PR TITLE
Add pipeline name to assertion messages using error_name

### PR DESCRIPTION
**Pull request by gasparitiago**

### CODE REVIEW

**Comment by gasparitiago** [comment]
@dsharletg I don't know if this solves the problem mentioned in https://github.com/halide/Halide/issues/5862 but, again, giving a try to, at least, improve the error messages.

**Comment by dsharletg** [comment]
I think this `error_name` is only used in a subset of the assertion messages.

@abadams what do you think? Should we just bite the bullet and add a pipeline name parameter to every `halide_error_*` function?

**Comment by abadams** [comment]
Yes, I think we need to bite that bullet, unfortunately. I don't see a good way around it. We'd have to plumb the pipeline name through in a lot of places in lowering too. Ugh.

File: test/generator/msan_aottest.cpp [hunk 1]
```diff
@@ -194,7 +194,7 @@ extern "C" int halide_msan_annotate_memory_is_initialized(void *user_context, co
         }
     } else if (annotate_stage == AnnotateIntermediateBuffer) {
         if (expect_intermediate_buffer_error) {
-            if (len != 80) {
+            if (len != 99) {
                 fprintf(stderr, "Failure: Expected error message of len=80, saw %d bytes\n", (unsigned int)len);
```
Comment (rootjalex) [inline test/generator/msan_aottest.cpp:198]: Error message should be updated to say len=99

**Comment by steven-johnson** [comment]
> Yes, I think we need to bite that bullet, unfortunately.

So should this PR proceed, or get withdrawn?

**Comment by steven-johnson** [comment]
> > Yes, I think we need to bite that bullet, unfortunately.
> 
> So should this PR proceed, or get withdrawn?

Monday morning review ping

**Comment by steven-johnson** [comment]
Is this PR still active?

**Comment by steven-johnson** [comment]
Closing this PR as there has been no activity in months.

**Pull request closed without merging**